### PR TITLE
Drop use of resultset decorator class in the ORM.

### PR DESCRIPTION
**Pull request by ADmad**
It serves no purpose anymore since ResultSetInterface simply extends CollectionInterface and ResultSet extends Collection.

### CODE REVIEW

**Review by othercorey — APPROVED** [review]
Looks ok to me

File: src/ORM/Association.php [hunk 1]
```diff
@@ -995,11 +995,11 @@ function (CollectionInterface $results, SelectQuery $query) use ($formatters, $p
                     $extracted[] = $result;
                 }
                 $extracted = $query->resultSetFactory()->createResultSet($extracted);
-                $decoratorClass = $query->resultSetFactory()->decoratorClass();
+                $resultSetClass = $query->resultSetFactory()->getResultSetClass();
                 foreach ($formatters as $callable) {
                     $extracted = $callable($extracted, $query);
                     if (!$extracted instanceof ResultSetInterface) {
-                        $extracted = new $decoratorClass($extracted);
+                        $extracted = new $resultSetClass($extracted);
```
Comment (markstory) [inline src/ORM/Association.php:1002]: If a result formatter returns a non-array value, wouldn't using `ResultSet` cause an error where as `ResultSetDecorator` would work?

I think skipping the decorator makes sense if the value is an array. For other iterable types though we should continue using the decorator or we could break userland code.

Comment (ADmad) [reply src/ORM/Association.php:1002]: On the `5.next` branch `ResultSet` and `ResultSetDecorator` are almost identical, both simply extend `Collection` (they only have slightly different implementation of `__debugInfo()`). 

So swapping one for another makes no practical difference. I can't foresee a case where dropping use of `ResultSetDecorator` could break userland code.

**Comment by jamisonbryant** [comment]
We were using it to cast something to a paginateable format a long time
ago. I can’t remember exactly what the source object was. But. I checked my
code base and we’re not doing it that way anymore.

File: src/ORM/Association.php (hunk 1 above)
Comment (markstory) [reply src/ORM/Association.php:1002]: :+1: I was looking at constructors on the 5.x branch 🤦

**Pull request merged**